SwrveSDK/src/main/java/com/swrve/sdk/QueuedEventsManager.java

```diff
@@ -0,0 +1,94 @@
+        final LinkedHashMap<Long, String> events = new LinkedHashMap<Long, String>();
+        if (!combinedEvents.isEmpty()) {
+            SwrveLogger.i(LOG_TAG, "Sending queued events");
+            try {
+                // Combine all events
+                Iterator<ILocalStorage> storageIt = combinedEvents.keySet().iterator();
+                while (storageIt.hasNext()) {
+                    events.putAll(combinedEvents.get(storageIt.next()));
+                }
+                eventsSent = events.size();
+                String data = com.swrve.sdk.EventHelper.eventsAsBatch(userId, appVersion, sessionToken, events, cachedLocalStorage);
+                SwrveLogger.i(LOG_TAG, "Sending " + events.size() + " events to Swrve");
+                postBatchRequest(data, new IPostBatchRequestListener() {
+                    public void onResponse(boolean shouldDelete) {
+                        if (shouldDelete) {
+                            // Remove events from where they came from
+                            Iterator<ILocalStorage> storageIt = combinedEvents.keySet().iterator();
+                            while (storageIt.hasNext()) {
+                                ILocalStorage storage = storageIt.next();
+                                storage.removeEventsById(combinedEvents.get(storage).keySet());
+                            }
+                        } else {
+                            SwrveLogger.e(LOG_TAG, "Batch of events could not be sent, retrying");
+                        }
+                    }
+                });
+            } catch (JSONException je) {
+                SwrveLogger.e(LOG_TAG, "Unable to generate event batch", je);
+            }
+        }
+        return eventsSent;
+    }
+
+    protected void postBatchRequest(String postData, final IPostBatchRequestListener listener) {
+
+        restClient.post(config.getEventsUrl() + SwrveBase.BATCH_EVENTS_ACTION, postData, new IRESTResponseListener() {
+            @Override
+            public void onResponse(RESTResponse response) {
+                boolean deleteEvents = true;
+                if (SwrveHelper.userErrorResponseCode(response.responseCode)) {
+                    SwrveLogger.e(LOG_TAG, "Error sending events to Swrve: " + response.responseBody);
+                } else if (SwrveHelper.successResponseCode(response.responseCode)) {
+                    SwrveLogger.i(LOG_TAG, "Events sent to Swrve");
+                } else if (SwrveHelper.serverErrorResponseCode(response.responseCode)) {
+                    deleteEvents = false;
+                    SwrveLogger.e(LOG_TAG, "Error sending events to Swrve: " + response.responseBody);
+                }
+
+                // Resend if we got a server error (5XX)
+                listener.onResponse(deleteEvents);
+            }
+
+            @Override
+            public void onException(Exception exp) {
```

Comment:
I know this was copy/paste but while we're in the neighbourhood, can we throw a SwrveLogger.e in here?